libs/zopengl/src/wrapper.zig

```diff
@@ -0,0 +1,2637 @@
+pub const BUFFER_MAPPED = bindings.BUFFER_MAPPED;
+pub const BUFFER_MAP_POINTER = bindings.BUFFER_MAP_POINTER;
+pub const STREAM_DRAW = bindings.STREAM_DRAW;
+pub const STREAM_READ = bindings.STREAM_READ;
+pub const STREAM_COPY = bindings.STREAM_COPY;
+pub const STATIC_DRAW = bindings.STATIC_DRAW;
+pub const STATIC_READ = bindings.STATIC_READ;
+pub const STATIC_COPY = bindings.STATIC_COPY;
+pub const DYNAMIC_DRAW = bindings.DYNAMIC_DRAW;
+pub const DYNAMIC_READ = bindings.DYNAMIC_READ;
+pub const DYNAMIC_COPY = bindings.DYNAMIC_COPY;
+pub const SAMPLES_PASSED = bindings.SAMPLES_PASSED;
+pub const SRC1_ALPHA = bindings.SRC1_ALPHA;
+
+// pub var genQueries: *const fn (n: Sizei, ids: [*c]Uint) callconv(.C) void = undefined;
+// pub var deleteQueries: *const fn (n: Sizei, ids: [*c]const Uint) callconv(.C) void = undefined;
+// pub var isQuery: *const fn (id: Uint) callconv(.C) Boolean = undefined;
+// pub var beginQuery: *const fn (target: Enum, id: Uint) callconv(.C) void = undefined;
+// pub var endQuery: *const fn (target: Enum) callconv(.C) void = undefined;
+// pub var getQueryiv: *const fn (target: Enum, pname: Enum, params: [*c]Int) callconv(.C) void = undefined;
+// pub var getQueryObjectiv: *const fn (id: Uint, pname: Enum, params: [*c]Int) callconv(.C) void = undefined;
+// pub var getQueryObjectuiv: *const fn (id: Uint, pname: Enum, params: [*c]Uint) callconv(.C) void = undefined;
+
+// pub var bindBuffer: *const fn (target: Enum, buffer: Uint) callconv(.C) void = undefined;
+pub fn bindBuffer(target: BufferTarget, buffer: Buffer) void {
+    assert(@bitCast(Uint, buffer) > 0);
+    bindings.bindBuffer(@enumToInt(target), @bitCast(Uint, buffer));
+}
+
+// pub var deleteBuffers: *const fn (n: Sizei, buffers: [*c]const Uint) callconv(.C) void = undefined;
+
+// pub var genBuffers: *const fn (n: Sizei, buffers: [*c]Uint) callconv(.C) void = undefined;
+pub fn genBuffers(ptr_or_slice: anytype) void {
+    const T = @TypeOf(ptr_or_slice);
+    const type_info = @typeInfo(T);
+    if (type_info != .Pointer) {
+        @compileError("genBuffers expects a single-item pointer or a slice");
+    }
+    if (type_info.Pointer.child != Buffer) {
+        @compileError("genBuffers expects pointer child type to be Buffer");
+    }
+    switch (type_info.Pointer.size) {
+        .One => {
+            bindings.genBuffers(1, @ptrCast([*c]Uint, ptr_or_slice));
+        },
+        .Slice => {
+            bindings.genBuffers(ptr_or_slice.len, @ptrCast([*c]Uint, ptr_or_slice.ptr));
+        },
+        else => {
+            @compileError("genBuffers expects a single-item pointer or a slice");
+        },
+    }
+}
+
+// pub var isBuffer: *const fn (buffer: Uint) callconv(.C) Boolean = undefined;
+
+// pub var bufferData: *const fn (
+//     target: Enum,
+//     size: Sizeiptr,
+//     data: ?*const anyopaque,
+//     usage: Enum,
+// ) callconv(.C) void = undefined;
+pub fn bufferData(
+    target: BufferTarget,
+    size: usize,
+    bytes: ?[*]const u8,
```

Comment:
I think we should pass an optional slice here?

Reply:
I added some asserts also. Note that the optional slice has to be unwrapped. What do you think?